pandas/core/sparse/series.py

```diff
@@ -65,142 +70,107 @@ class SparseSeries(Series):
-
-            elif isinstance(data, (Series, dict)):
-                data = Series(data, index=index)
-                index = data.index.view()
-
-                res = make_sparse(data, kind=kind, fill_value=fill_value)
-                data, sparse_index, fill_value = res
-
-            elif isinstance(data, (tuple, list, np.ndarray)):
-                # array-like
-                if sparse_index is None:
-                    res = make_sparse(data, kind=kind, fill_value=fill_value)
-                    data, sparse_index, fill_value = res
-                else:
-                    assert (len(data) == sparse_index.npoints)
-
-            elif isinstance(data, SingleBlockManager):
-                if dtype is not None:
-                    data = data.astype(dtype)
-                if index is None:
-                    index = data.index.view()
-                elif not data.index.equals(index) or copy:  # pragma: no cover
-                    # GH#19275 SingleBlockManager input should only be called
-                    # internally
-                    raise AssertionError('Cannot pass both SingleBlockManager '
-                                         '`data` argument and a different '
-                                         '`index` argument.  `copy` must '
-                                         'be False.')
-
-            else:
-                length = len(index)
-
-                if data == fill_value or (isna(data) and isna(fill_value)):
-                    if kind == 'block':
-                        sparse_index = BlockIndex(length, [], [])
-                    else:
-                        sparse_index = IntIndex(length, [])
-                    data = np.array([])
-
-                else:
-                    if kind == 'block':
-                        locs, lens = ([0], [length]) if length else ([], [])
-                        sparse_index = BlockIndex(length, locs, lens)
-                    else:
-                        sparse_index = IntIndex(length, index)
-                    v = data
-                    data = np.empty(length)
-                    data.fill(v)
-
-            if index is None:
-                index = ibase.default_index(sparse_index.length)
-            index = ensure_index(index)
-
-            # create/copy the manager
-            if isinstance(data, SingleBlockManager):
-
-                if copy:
-                    data = data.copy()
-            else:
-
-                # create a sparse array
-                if not isinstance(data, SparseArray):
-                    data = SparseArray(data, sparse_index=sparse_index,
-                                       fill_value=fill_value, dtype=dtype,
-                                       copy=copy)
-
-                data = SingleBlockManager(data, index)
-
-        generic.NDFrame.__init__(self, data)
-
-        self.index = index
-        self.name = name
+        # TODO: Most of this should be refactored and shared with Series
```

Comment:
or rather deprecate instead of putting effort in refactoring / sharing code with Series ?

Reply:
You mean deprecate SparseSeries entirely? Yes, that seems like the best path forward.